pkg/registry/deployment/etcd/etcd_test.go

```diff
@@ -180,73 +185,69 @@ func TestWatch(t *testing.T) {
 	)
 }
 
-// TODO(madhusudancs): Fix this when Scale group issues are resolved  (see issue #18528).
-
-// func validNewScale() *extensions.Scale {
-// 	return &extensions.Scale{
-// 		ObjectMeta: api.ObjectMeta{Name: name, Namespace: namespace},
-// 		Spec: extensions.ScaleSpec{
-// 			Replicas: validDeployment.Spec.Replicas,
-// 		},
-// 		Status: extensions.ScaleStatus{
-// 			Replicas: validDeployment.Status.Replicas,
-// 			Selector: validDeployment.Spec.Template.Labels,
-// 		},
-// 	}
-// }
-
-// var validScale = *validNewScale()
-
-// func TestScaleGet(t *testing.T) {
-// 	storage, server := newStorage(t)
-// 	defer server.Terminate(t)
-
-// 	ctx := api.WithNamespace(api.NewContext(), namespace)
-// 	key := etcdtest.AddPrefix("/deployments/" + namespace + "/" + name)
-// 	if err := storage.Deployment.Storage.Set(ctx, key, &validDeployment, nil, 0); err != nil {
-// 		t.Fatalf("unexpected error: %v", err)
-// 	}
-
-// 	expect := &validScale
-// 	obj, err := storage.Scale.Get(ctx, name)
-// 	if err != nil {
-// 		t.Fatalf("unexpected error: %v", err)
-// 	}
-// 	scale := obj.(*extensions.Scale)
-// 	if e, a := expect, scale; !api.Semantic.DeepDerivative(e, a) {
-// 		t.Errorf("unexpected scale: %s", util.ObjectDiff(e, a))
-// 	}
-// }
-
-// func TestScaleUpdate(t *testing.T) {
-// 	storage, server := newStorage(t)
-// 	defer server.Terminate(t)
-
-// 	ctx := api.WithNamespace(api.NewContext(), namespace)
-// 	key := etcdtest.AddPrefix("/deployments/" + namespace + "/" + name)
-// 	if err := storage.Deployment.Storage.Set(ctx, key, &validDeployment, nil, 0); err != nil {
-// 		t.Fatalf("unexpected error: %v", err)
-// 	}
-// 	replicas := 12
-// 	update := extensions.Scale{
-// 		ObjectMeta: api.ObjectMeta{Name: name, Namespace: namespace},
-// 		Spec: extensions.ScaleSpec{
-// 			Replicas: replicas,
-// 		},
-// 	}
-
-// 	if _, _, err := storage.Scale.Update(ctx, &update); err != nil {
-// 		t.Fatalf("unexpected error: %v", err)
-// 	}
-// 	obj, err := storage.Deployment.Get(ctx, name)
-// 	if err != nil {
-// 		t.Fatalf("unexpected error: %v", err)
-// 	}
-// 	deployment := obj.(*extensions.Deployment)
-// 	if deployment.Spec.Replicas != replicas {
-// 		t.Errorf("wrong replicas count expected: %d got: %d", replicas, deployment.Spec.Replicas)
-// 	}
-// }
+func TestScaleGet(t *testing.T) {
+	storage, server := newStorage(t)
+	defer server.Terminate(t)
+
+	ctx := api.WithNamespace(api.NewContext(), namespace)
+	key := etcdtest.AddPrefix("/deployments/" + namespace + "/" + name)
+	if err := storage.Deployment.Storage.Set(ctx, key, &validDeployment, nil, 0); err != nil {
```

Comment:
That looks nice, but isn't widely used in the code yet, so I wouldn't block on that.